addons/point_of_sale/static/src/app/components/popups/cash_move_popup/cash_move_popup.js

```diff
@@ -90,10 +95,16 @@ export class CashMovePopup extends Component {
             ? this.env.utils.formatCurrency(parseFloat(value))
             : "";
     }
-    _prepareTryCashInOutPayload(type, amount, reason, extras) {
-        return [[this.pos.session.id], type, amount, reason, extras];
+    _prepareTryCashInOutPayload(type, amount, reason, partnerId, extras) {
+        return [[this.pos.session.id], type, amount, reason, partnerId, extras];
     }
     isValidCashMove() {
         return this.env.utils.isValidFloat(this.state.amount) && this.state.reason.trim() !== "";
     }
+    async openDetails() {
+        const cashMoves = await this.pos.data.call("pos.session", "get_cash_in_out_list", [
+            this.pos.session.id,
+        ]);
+        this.dialog.add(CashMoveListPopup, { cashMoves: cashMoves, partnerId: this.partnerId });
```

Comment:
```suggestion
        this.dialog.add(CashMoveListPopup, { cashMoves: cashMoves.map((m) => ({ ..., "date": DateTime.fromSQL(m.date)}), partnerId: this.partnerId });
```

do the preprocessing here. You can remove the `get cashMoves` in the new popup.